app/src/main/java/com/babylon/wallet/android/utils/ResultExtensions.kt

```diff
@@ -0,0 +1,9 @@
+package com.babylon.wallet.android.utils
+
+import com.babylon.wallet.android.domain.RadixWalletException
+import com.babylon.wallet.android.domain.asRadixWalletException
+
+inline fun <T> Result<T>.onFailureWithRadixException(action: (exception: RadixWalletException) -> Unit): Result<T> {
```

Comment:
I completely understand why we need this and I am ok with it, but I also have the concern that it may become overused. Most of our errors implement RadixWalletException, therefore we will tend to use this function that is convenient and lets us directly map an error to RadixWalletException. But what about the rest of the exceptions that may have propagated all the way back to the stack. These errors will be ignored. Currently I see many cases were we do ignore them. 

So personally I would keep the `asRadixWalletException()` but would avoid turning a result directly to this kind of exception.